src/service/cli/routes/offers.e2e.test.js

```diff
@@ -0,0 +1,185 @@
+'use strict';
+
+const request = require(`supertest`);
+const find = require(`lodash/find`);
+
+const server = require(`./server`);
+const {initializeOffersDatabase, clearOffersDatabase, mockData} = require(`../../utils/prepareDataForTests`);
+
+const offerKeys = [`id`, `title`, `picture`, `description`, `type`, `sum`, `сategory`, `comments`];
+
+const postOfferData = {
+  "title": "Куплю антиквариат.",
+  "picture": "item00.jpg",
+  "description": [
+    "Кажется, что это хрупкая вещь.",
+    "При покупке с меня бесплатная доставка в черте города."
+  ],
+  "type": "offer",
+  "sum": 4667,
+  "сategory": [
+    "Посуда"
+  ],
+  "comments": [
+    {
+      "id": "BW1qOk",
+      "text": "Оплата наличными или перевод на карту?"
+    },
+    {
+      "id": "DDgAGS",
+      "text": {
+        "text": "tesaasd"
+      }
+    },
+    {
+      "id": "kAOd7a",
+      "text": "tesaasd"
+    }
+  ]
+};
+
+const putOfferData = {
+  "id": "1",
+  "title": "Куплю антиквариат.",
+  "picture": "item00.jpg",
+  "description": [
+    "Кажется, что это хрупкая вещь.",
+    "При покупке с меня бесплатная доставка в черте города."
+  ],
+  "type": "offer",
+  "sum": 4667,
+  "сategory": [
+    "Посуда"
+  ],
+  "comments": [
+    {
+      "id": "BW1qOk",
+      "text": "Оплата наличными или перевод на карту?"
+    },
+    {
+      "id": "DDgAGS",
+      "text": {
+        "text": "tesaasd"
+      }
+    },
+    {
+      "id": "kAOd7a",
+      "text": "tesaasd"
+    }
+  ]
+};
+
+const getOfferData = {
+  "id": "2",
+  "title": "Продам Куплю новую приставку Sony Playstation 5.",
+  "picture": "item14.jpg",
+  "description": [
+    "Товар в отличном состоянии.",
+    "Это настоящая находка для коллекционера!"
+  ],
+  "type": "offer",
+  "sum": 69196,
+  "сategory": [
+    "Журналы",
+    "Разное"
+  ],
+  "comments": [
+    {
+      "id": "UFDHfT",
+      "text": "А сколько игр в комплекте?"
+    },
+    {
+      "id": "tMJGgq",
+      "text": "С чем связана продажа? Почему так дешёво?"
+    },
+    {
+      "id": "LRFSlR",
+      "text": "Неплохо, но дорого"
+    }
+  ]
+};
+
+describe(`Offers API end-points`, () => {
+  beforeAll(() => {
+    initializeOffersDatabase();
+  });
+
+  afterAll(() => {
+    clearOffersDatabase();
+  });
+
+  test(`When GET offers status code should be 200`, async () => {
+    const res = await request(server).get(`/offers`);
+    expect(res.statusCode).toBe(200);
+  });
+
+  test(`When GET offer by ID data should be contain following keys`, async () => {
+    const res = await request(server).get(`/offers/1`);
+    expect(res.statusCode).toBe(200);
+    offerKeys.forEach(key => expect(res.body).toHaveProperty(key));
```

Comment:
Проверяем не только ключи, а ещё и данные
```suggestion
    expect(res.body).toBeEq(offersFixture['1']);
```